pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -338,6 +356,71 @@ func (f *framework) RunScorePlugins(pc *PluginContext, pod *v1.Pod, nodes []*v1.
 	return pluginToNodeScoreMap, nil
 }
 
+// RunNormalizeScorePlugins runs the normalize score plugins. It should be called after
+// RunScorePlugins with the PluginToNodeScoreMap result. It then modifies the map with
+// normalized scores. It returns a non-success Status if any of the normalize score plugins
+// returns a non-success status.
+func (f *framework) RunNormalizeScorePlugins(pc *PluginContext, pod *v1.Pod, scores PluginToNodeScoreMap) *Status {
+	ctx, cancel := context.WithCancel(context.Background())
+	errCh := schedutil.NewErrorChannel()
+	workqueue.ParallelizeUntil(ctx, 16, len(f.normalizeScorePlugins), func(index int) {
+		pl := f.normalizeScorePlugins[index]
+		nodeScoreList, ok := scores[pl.Name()]
+		// During framework initialization, we make sure that NormalizeScore plugins are
+		// a subset of Score plugins. So nodeScoreList should exist given that implementation
+		// of RunScorePlugins produce the correct output.
+		if !ok {
+			err := fmt.Errorf("NormalizeScore plugin %v has no corresponding scores in the PluginToNodeScoreMap.", pl.Name())
+			errCh.SendErrorWithCancel(err, cancel)
+		}
+		status := pl.NormalizeScore(pc, nodeScoreList)
+		if !status.IsSuccess() {
+			err := fmt.Errorf("normalize score plugin %v failed with error %v", pl.Name(), status.Message())
+			errCh.SendErrorWithCancel(err, cancel)
+		}
+	})
+
+	if err := errCh.ReceiveError(); err != nil {
+		msg := fmt.Sprintf("error while running normalize score plugin for pod %v: %v", pod.Name, err)
+		klog.Error(msg)
+		return NewStatus(Error, msg)
+	}
+
+	return nil
+}
+
+// ApplyScoreWeights applies weights to the score results. It should be called after
+// RunNormalizeScorePlugins.
+func (f *framework) ApplyScoreWeights(pc *PluginContext, pod *v1.Pod, scores PluginToNodeScoreMap) *Status {
+	ctx, cancel := context.WithCancel(context.Background())
+	errCh := schedutil.NewErrorChannel()
+	workqueue.ParallelizeUntil(ctx, 16, len(f.scorePlugins), func(index int) {
+		pl := f.scorePlugins[index]
+		weight, weightExists := f.pluginNameToWeightMap[pl.Name()]
+		if !weightExists {
+			err := fmt.Errorf("weight does not exist for plugin %v", pl.Name())
+			errCh.SendErrorWithCancel(err, cancel)
+			return
+		}
+		nodeScoreList, ok := scores[pl.Name()]
+		if !ok {
+			err := fmt.Errorf("NormalizeScore plugin %v has no corresponding scores in the PluginToNodeScoreMap.", pl.Name())
```

Comment:
nit: s/NormalizeScore/normalize score/
nit: remove the `.` in the end

errors should start with lower case letter